File: C++_Python_OpenCV/OpenCV_04_Rotation.cpp
# include <opencv2/opencv.hpp>
# include <iostream>

int main (){
	// Reading and showing the demo image
	cv::Mat img = cv::imread("demo_image.png", cv::IMREAD_COLOR);
	cv::imshow("Demo image", img);
	
	// Image Rotation
	// Angle: 45 degrees
	// Calculating the center of the image
	cv::Point2f img_center((img.cols - 1) / 2.0, (img.rows - 1) / 2.0);
	// Rotating the matrix by 45 degrees
	cv::Mat rotation_matix = cv::getRotationMatrix2D(img_center, 45, 1.0);
	// Rotating the image using warpAffine
	cv::Mat img_rot;
	cv::warpAffine(img, img_rot, rotation_matix, img.size());
	// Showing the rotated image
	cv::imshow("Rotated image", img_rot);
	
	// Waiting for key 0 to destroy all windows
	cv::waitKey(0);
	cv::destroyAllWindows();
	
	return 0;
}

File: C++_Python_OpenCV/OpenCV_04_Rotation.py
import cv2

# Reading and showing the demo image 
img = cv2.imread('demo_image.png', cv2.IMREAD_COLOR)
cv2.imshow('Demo image', img)

# Image Rotation:
# Angle: 45 degrees
# Calculating the center of the image
img_height = img.shape[0]
img_width = img.shape[1]
img_center = (img_width/2, img_height/2)
# Rotating the matrix by 45 degrees
rotation_matix = cv2.getRotationMatrix2D(img_center, 45, scale=1)
# Rotating the image using cv2.warpAffine
img_rot = cv2.warpAffine(src=img, M=rotation_matix, dsize=(img_width, img_height))
# Showing the rotated image
cv2.imshow('Rotated image', img_rot)

# Waiting for key 0 to destroy all windows
cv2.waitKey(0)
cv2.destroyAllWindows()
